### ISSUE TITLE
UI: Fix Locked "Override Root Disk Size" switch

### ISSUE THREAD

**Issue by GabrielBrascher** [apache/cloudstack#5842]
##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
UI
~~~

##### CLOUDSTACK VERSION

~~~
4.15.2 (maybe on a previous 4.15 as well)
~~~

##### OS / ENVIRONMENT

N/A

##### SUMMARY

Once the lock 

##### STEPS TO REPRODUCE

~~~
1. Deploy a VM on UI
2. During the deployment phase,  choose an offering that does not allow to override the disk
3. choose back an offering that does allow to override disk size
4. button is already stuck and do not switch back
~~~

![image](https://user-images.githubusercontent.com/5025148/148546886-083ab3cd-6d15-47af-b682-b967a94237fb.png)

##### EXPECTED RESULTS

~~~
Override Disk toggle button should be unblocked as soon as an offering that allows custom root disk size.
~~~

##### ACTUAL RESULTS

~~~
Once an offering that has a root disk size set is chosen, the UI locks and do not change back, regardless of the offering chosen afterward.
~~~
![image](https://user-images.githubusercontent.com/5025148/148546246-d5cfd9ae-9f90-4fea-a3da-a3ed26bfc3dc.png)

**Comment by DaanHoogland** [comment]
@GabrielBrascher i think this is fixed in 4.16, can you try?

**Comment by GabrielBrascher** [comment]
@DaanHoogland It still failed on a test I did on the main branch (4.17.0-snapshot).
Unless I am missing a specific commit from my recent build package/branch.

I remember I've fixed  an issue regarding the Root disk size not being presented correctly on the VM deployment at #5796 
But this is a separate issue that still needs to be addressed.

**Comment by DaanHoogland** [comment]
ok, I tested in a 4.16 environment and couldn't reproduce. Of course this can be due to not understanding the scenario;
I added a constrained offering (maybe you can give me the specifics?)
I started the deploy wizard.
I changed to the constrained offering and the custom root size switch remained active.
I could change back to a standard offering as well, but that action seems mute, given the above.

**Comment by GabrielBrascher** [comment]
@DaanHoogland the issue appears only if there is at least one offering with the Root disk size set.

When the offering is selected, then the switch button for override the disk size should be blocked. the problem is that the button does not get back into the "unblocked" state once you choose another offering that does not set a root disk size.

Please let me know if I did not make me clear. I can also show a bit more details on how to reproduce or get a few screenshots.

**Comment by DaanHoogland** [comment]
> Please let me know if I did not make me clear. I can also show a bit more details on how to reproduce or get a few screenshots.

clear @GabrielBrascher I interpreted a offering with a constraint root-size (but not fixed) would already trigger the problem.
thanks

**Comment by sureshanaparti** [comment]
Fixed in #5843
